lib/membrane/pipeline.ex

```
@@ -295,43 +295,69 @@ defmodule Membrane.Pipeline do
   end
 
   @doc """
-  Gracefully terminates the pipeline.
-
-  Accepts two options:
-    * `blocking?` - tells whether to stop the pipeline synchronously
-    * `timeout` - if `blocking?` is set to true it tells how much
-      time (ms) to wait for pipeline to get terminated. Defaults to 5000.
+  Terminates the pipeline.
+
+  Accepts three options:
+    * `asynchronous?` - if set to `true`, pipline termination won't be blocking and
+      will be executed in the process, which pid is returned as function result. If
+      set to `false`, pipeline termination will be blocking and will be executed in
+      the process, that called this function.
+    * `timeout` - tells how much time (ms) to wait for pipeline to get gracefully
+      terminated. Defaults to 5000.
+    * `force?` - if set to `true` and pipeline is still alive after `timeout`,
+      pipeline will be killed using `Process.exit/2` with second argument set to
+      `:kill`, and function will return `{:error, :timeout}`. If set to `false` and
+      pipeline is still alive after `timeout`, function will raise an error.
```

[SOME THOUGHTS] Hmm should it raise under these circumstances? I think it would be better if we have simply returned the `:error` tuple, and let the user raise on her/his own